### PR TITLE
feat: integrate ThemeToggle into app shell sidebar and auth layout (Phase 7.3)

### PR DESCRIPTION
**Pull request by JakubAnderwald**
## Summary
- Add ThemeToggle component to the sidebar footer in the app shell layout
- Add ThemeToggle component to the top-right corner of the auth layout
- Make useTheme hook defensive against missing localStorage/matchMedia in test environments
- Add useTheme mock to app-shell integration tests

## Test plan
- [x] Unit + integration tests pass (437/437)
- [x] E2E tests pass (42 passed, 1 flaky pre-existing)
- [x] ESLint: 0 errors
- [x] TypeScript: clean
- [x] Build: passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Added theme toggle component to the authentication and app layouts for convenient theme switching.

* **Bug Fixes**
  * Improved error handling for theme storage to prevent failures in restricted browser environments.

* **Tests**
  * Enhanced integration test stability with proper theme hook mocking.

* **Documentation**
  * Updated UI redesign plan progress tracker.

### CODE REVIEW

File: src/hooks/use-theme.ts [hunk 1]
```diff
@@ -9,8 +9,12 @@ const STORAGE_KEY = "theme";
...
 function getStoredTheme(): Theme {
   if (typeof window === "undefined") return "system";
-  const stored = localStorage.getItem(STORAGE_KEY);
-  if (stored === "light" || stored === "dark" || stored === "system") return stored;
+  try {
+    const stored = localStorage.getItem(STORAGE_KEY);
+    if (stored === "light" || stored === "dark" || stored === "system") return stored;
+  } catch {
+    // localStorage may not be available in some environments
+  }
```
[the remark this answers is not in the document]
Comment (JakubAnderwald) [reply src/hooks/use-theme.ts]: Fixed in 9fa97f3 — added `typeof window.matchMedia \!== "function"` guards to both `getSystemTheme()` and the media query listener effect.

Regarding the Sentry suggestion for the catch blocks: these catch blocks handle edge cases where localStorage is genuinely unavailable (e.g., private browsing, test environments). Logging to Sentry would create noise for expected behavior. The fallback to `"system"` is the correct behavior here.
